utils/mosaic_processor.py

```
@@ -33,17 +33,72 @@
+        utils_dir = Path(__file__).parent
+        monitor_script = utils_dir / "mosaic_progress_display.py"
+
+        if not monitor_script.exists():
+            logger.warning(f"Monitor script not found: {monitor_script}", indent=4)
+            return None
+
+        # Get Python executable (try to use same environment as current process)
+        python_exe = "python"  # Default fallback
+        try:
+            python_exe = sys.executable
+        except:
```

Ruff (0.14.2)

Do not use bare `except`

(E722)